src/o365/graph/commands/teams/teams-set.ts

```diff
@@ -0,0 +1,171 @@
+    });
+    return telemetryProps;
+  }
+
+  private mapRequestBody(options: Options): any {
+    const requestBody: any = {};
+    if (options.displayName) {
+      requestBody.displayName = options.displayName;
+    }
+    if (options.description) {
+      requestBody.description = options.description;
+    }
+    if (options.mailNickName) {
+      requestBody.mailNickName = options.mailNickName;
+    }
+    if (options.classification) {
+      requestBody.classification = options.classification;
+    }
+    if (options.visibility) {
+      requestBody.visibility = options.visibility;
+    }
+    return requestBody;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): Promise<{}> => {
+        const body: any = this.mapRequestBody(args.options);
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/beta/groups/${encodeURIComponent(args.options.teamId)}`,
+          headers: {
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          },
+          body: body,
+          json: true
+        };
+
+        return request.patch(requestOptions);
+      })
+      .then((): void => {
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The ID of the Microsoft Teams team for which to update settings'
+      },
+      {
+        option: '--displayName [displayName]',
+        description: 'The display name for the Microsoft Teams team'
+      },
+      {
+        option: '--description [description]',
+        description: 'The description for the Microsoft Teams team'
+      },
+      {
+        option: '--mailNickName [mailNickName]',
+        description: 'The mail alias for the Microsoft Teams team'
+      },
+      {
+        option: '--classification [classification]',
+        description: 'The classification for the Microsoft Teams team with valid values HBI, MBI, LBI, GDPR'
+      },
+      {
+        option: '--visibility [visibility]',
+        description: 'Specifies the visibility of the Microsoft Teams team with valid values Private,Public'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.teamId) {
+        return 'Required parameter teamId missing';
+      }
+
+      if (!Utils.isValidGuid(args.options.teamId)) {
+        return `${args.options.teamId} is not a valid GUID`;
+      }
+      return true;
```

Comment:
We should add validating the value of the `visibility` option if specified.

Reply:
Sure thanks @waldekmastykarz will add it